src/main/java/org/distributed/systems/chord/actors/Node.java

```diff
@@ -61,80 +76,70 @@ public static Props props(ActorRef manager) {
     public void preStart() throws Exception {
         super.preStart();
         log.info("Starting up...     ref: " + getSelf());
+        this.id = getNodeId();
 
-        // Generating Id:
-//        Random randomGenerator = new Random();
-//        long randomInt = randomGenerator.nextInt(ChordStart.M);
-        long envVal;
-        HashUtil hashUtil = new HashUtil();
-        if (System.getenv("NODE_ID") == null) {
-            String hostName = config.getString("akka.remote.artery.canonical.hostname");
-            String port = config.getString("akka.remote.artery.canonical.port");
-              // FIXME Should be IP
-            envVal = Math.floorMod(hashUtil.hash(hostName + ":" + port), AMOUNT_OF_KEYS);
-        } else {
-            envVal = Long.parseLong(System.getenv("NODE_ID"));
-        }
-
-        this.id = envVal;
-        System.out.println("Node Id " + this.id);
+        // Init the FingerTable
+        fingerTable = new FingerTableEntry[Node.m];
 
         if (this.type.equals("central")) {
-            this.predecessor = getSelf();
-            this.predecessorId = this.id;
-            this.sucessor = getSelf();
-            this.sucessorId = this.id;
-            System.out.println("Started As Central Node");
+            this.predecessor = null;
+            this.predecessorId = -1;
+            this.setSuccessor(getSelf(), this.id);
+            System.out.println("Bootstrapped Central Node");
+            getSelf().tell(new Stabelize.Request(), getSelf());
         } else {
-            System.out.println("Started As Regular Node");
             final String centralNodeAddress = getCentralNodeAddress();
             Timeout timeout = Timeout.create(Duration.ofMillis(ChordStart.STANDARD_TIME_OUT));
-            System.out.println("Looking For Central Node");
             Future<ActorRef> centralNodeFuture = getContext().actorSelection(centralNodeAddress).resolveOne(timeout);
-            this.centralNode = (ActorRef) Await.result(centralNodeFuture, timeout.duration());
-            System.out.println("Found Central Node");
+            ActorRef centralNode = (ActorRef) Await.result(centralNodeFuture, timeout.duration());
```

Comment:
I still am not really sure about Await.result and such, but if it works it works

Reply:
Yeah it worked, so i did not remove it. But we can experiement I guess, because blocking really fucket up sometime. Like timeouts etc.

Reply:
:+1: for a task on trello